sbt-converter/src/main/scala/org/scalablytyped/converter/plugin/RemoteCache.scala

```diff
@@ -14,4 +17,19 @@ object RemoteCache {
     * The "directory" above will contain two folders after sync: `runs` and `org.scalablytyped`
     */
   case class Rsync(push: String, pull: URI) extends RemoteCache
+
+  case class S3(
+      push: S3.PushConfiguration,
+      pull: URI,
+  ) extends RemoteCache
+
+  object S3 {
+    case class PushConfiguration(
+        bucket:              String,
+        prefix:              Option[String] = None,
+        region:              Option[Region] = None,
```

Comment:
I don't know what these AWS types are and do, but would it make sense to replace them with just strings say? I think it's not a good user experience for the user of a build script to import things, and it would preclude us from swapping implementation down the line

Reply:
It would be possible for the Region (even though as it's an Enum I like the additional compiletime typesafety here, but a .valueOf would work just as fine).
It's not possible for the CredentialsProvider, except we force to use the Static one, which might not always be preferred as there might be an AWS auth setup already using the AWS-Preferred Default Credential Provider ( https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/credentials.html ).

We could try to do both though by e.g. using a builder (which would get rid of those Optionals too), like:

```scala
import java.net.URI

import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, AwsCredentialsProvider, StaticCredentialsProvider}
import software.amazon.awssdk.regions.Region

case class S3(
    bucket:              String,
    prefix:              Option[String],
    region:              Option[Region],
    endpointOverride:    Option[URI],
    credentialsProvider: Option[AwsCredentialsProvider],
) {
  def withPrefix(prefix:     String) = copy(prefix           = Some(prefix))
  def withEndpoint(endpoint: String) = copy(endpointOverride = Some(new URI(endpoint)))

  def withRegion(region:    String) = copy(region = Some(Region.of(region)))
  def withAwsRegion(region: Region) = copy(region = Some(region))

  def withStaticCredentials(accessKey: String, secretKey: String) =
    copy(credentialsProvider = Some(
      StaticCredentialsProvider.create(
        AwsBasicCredentials.create(
          accessKey,
          secretKey,
        ),
      ),
    ),
    )
  def withAwsCredentialsProvider(awsCredentialsProvider: AwsCredentialsProvider) =
    copy(credentialsProvider = Some(awsCredentialsProvider))
}

object S3 {
  def apply(bucket: String): S3 = S3(bucket, None, None, None, None)

}

// usage
S3("myBucket")
  .withStaticCredentials("myAccessKey", "mySecretKey")
  .withRegion("myStringRegion")
```

So you could configure it without any explicit AWS models, except you want to as you use AWS anyway and need an AwsCredentialsProvider

Reply:
Just another random proposal: Would it make sense to pick up config files like`~/.aws/config` instead for authentication?

Reply:
That builder thing looks great to me by the way

Reply:
> Just another random proposal: Would it make sense to pick up config files like`~/.aws/config` instead for authentication?

That's what the DefaultCredentialsProvider does (which is picked when you don't set one explicitly). 

> That builder thing looks great to me by the way

Great, I'll make the changes